bokehjs/src/lib/core/property_mixins.ts

```diff
@@ -223,190 +223,25 @@ export const TextVector: p.DefineOf<TextVector> = {
-  band_hatch_alpha: Hatch["hatch_alpha"]
-  band_hatch_scale: Hatch["hatch_scale"]
-  band_hatch_pattern: Hatch["hatch_pattern"]
-  band_hatch_weight: Hatch["hatch_weight"]
-  band_hatch_extra: Hatch["hatch_extra"]
-}
-
-export type OutlineLine = {
-  outline_line_color: Line["line_color"]
-  outline_line_alpha: Line["line_alpha"]
-  outline_line_width: Line["line_width"]
-  outline_line_join: Line["line_join"]
-  outline_line_cap: Line["line_cap"]
-  outline_line_dash: Line["line_dash"]
-  outline_line_dash_offset: Line["line_dash_offset"]
-}
-
-export type BorderFill = {
-  border_fill_color: Fill["fill_color"]
-  border_fill_alpha: Fill["fill_alpha"]
-}
-
-export type SeparatorLine = {
-  separator_line_color: Line["line_color"]
-  separator_line_alpha: Line["line_alpha"]
-  separator_line_width: Line["line_width"]
-  separator_line_join: Line["line_join"]
-  separator_line_cap: Line["line_cap"]
-  separator_line_dash: Line["line_dash"]
-  separator_line_dash_offset: Line["line_dash_offset"]
-}
-
-export type GroupText = {
-  group_text_color: Text["text_color"]
-  group_text_alpha: Text["text_alpha"]
-  group_text_font: Text["text_font"]
-  group_text_font_size: Text["text_font_size"]
-  group_text_font_style: Text["text_font_style"]
-  group_text_align: Text["text_align"]
-  group_text_baseline: Text["text_baseline"]
-  group_text_line_height: Text["text_line_height"]
-}
-
-export type SubGroupText = {
-  subgroup_text_color: Text["text_color"]
-  subgroup_text_alpha: Text["text_alpha"]
-  subgroup_text_font: Text["text_font"]
-  subgroup_text_font_size: Text["text_font_size"]
-  subgroup_text_font_style: Text["text_font_style"]
-  subgroup_text_align: Text["text_align"]
-  subgroup_text_baseline: Text["text_baseline"]
-  subgroup_text_line_height: Text["text_line_height"]
-}
+type Prefixed<P extends string, T> = {[key in keyof T & string as `${P}_${key}`]: T[key]}
+
+export type LabelText = Prefixed<"label", Text>
+export type InactiveFill = Prefixed<"inactive", Fill>
+export type BorderLine = Prefixed<"border", Line>
+export type BackgroundFill = Prefixed<"background", Fill>
+export type MajorLabelText = Prefixed<"major_label", Text>
+export type TitleText = Prefixed<"title", Text>
+export type MajorTickLine = Prefixed<"major_tick", Line>
+export type MinorTickLine = Prefixed<"minor_tick", Line>
+export type BarLine = Prefixed<"bar", Line>
+export type AxisLine = Prefixed<"axis", Line>
+export type AxisLabelText = Prefixed<"axis_label", Text>
+export type GridLine = Prefixed<"grid", Line>
+export type MinorGridLine = Prefixed<"minor_grid", Line>
+export type BandFill = Prefixed<"band", Fill>
+export type BandHatch = Prefixed<"band", Hatch>
+export type OutlineLine = Prefixed<"outline", Line>
+export type BorderFill = Prefixed<"border", Fill>
+export type SeparatorLine = Prefixed<"separator", Line>
+export type GroupText = Prefixed<"group", Text>
+export type SubGroupText = Prefixed<"subgroup", Text>
```

Comment:
This is fantastic. Only suggest now that they are one-liners to sort alphabetically